mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -277,39 +277,38 @@ object MatrixFactorizationModel extends Loader[MatrixFactorizationModel] {
       num: Int): RDD[(Int, Array[(Int, Double)])] = {
     val srcBlocks = blockify(rank, srcFeatures)
     val dstBlocks = blockify(rank, dstFeatures)
-    val ratings = srcBlocks.cartesian(dstBlocks).flatMap {
-      case (users, items) =>
-      val m = users.size
-      val n = math.min(items.size, num)
+    val ratings = srcBlocks.cartesian(dstBlocks).flatMap { case (srcIter, dstIter) =>
+      val m = srcIter.size
+      val n = math.min(dstIter.size, num)
       val output = new Array[(Int, (Int, Double))](m * n)
       var j = 0
-      users.foreach (user => {
+      srcIter.foreach { case (srcId, srcFactor) =>
           def order(a: (Int, Double)) = a._2
           val pq: BoundedPriorityQueue[(Int, Double)] =
             new BoundedPriorityQueue[(Int, Double)](n)(Ordering.by(order))
```

Comment:
I believe you can just do `Ordering.by(_._2)` without needing to define `def order(...` above